jdaviz/__init__.py

```diff
@@ -26,14 +26,91 @@
            'viewers',
            'new_viewers',
            'get_data']
-_incl = ['App', 'enable_hot_reloading', '__version__']
+_incl = ['enable_hot_reloading', '__version__', 'gca', 'get_all_apps', 'new']
 _temporary_incl = ['open', 'Cubeviz', 'Imviz', 'Mosviz', 'Rampviz', 'Specviz', 'Specviz2d']
 __all__ = _expose + _incl + _temporary_incl
 
 
-global _ca
+global _apps
+global _current_index
 
-_ca = App(api_hints_obj='jd')
+_apps = []
+
+
+def new(replace=False, set_as_current=True):
+    """
+    Create a new jdaviz application instance and assign as the current instance.
+
+    Parameters
+    ----------
+    replace : bool, optional
+        If True, replaces the current application instance with the new one.
+        Default is False, which means a new instance is added to the list of applications.
+    set_as_current : bool, optional
+        If True, sets the newly created application instance as the current instance.
+        Default is True.
+
+    Returns
+    -------
+    App
+        A new instance of the App class.
+    """
+    global _apps
+    global _current_index
+    ca = App(api_hints_obj='jd')
+    if replace:
+        _apps[_current_index] = ca
+    else:
+        _apps += [ca]
+        if set_as_current:
+            _current_index = len(_apps) - 1
+    return ca
+
+
+new()
+
+
+def get_all_apps():
+    """
+    Get a list of all jdaviz application instances.
+
+    Returns
+    -------
+    list
+        A list of all jdaviz application instances.
+    """
+    return _apps
+
+
+def gca(index=None, set_as_current=True):
+    """
+    Get the current jdaviz application instance.
+
+    Parameters
+    ----------
+    index : int, optional
+        The index of the application instance to retrieve.
+        Default is the current instance.
+    set_as_current : bool, optional
+        If True, sets the application instance at the specified index as the current instance.
+        Default is True.
+
+    Returns
+    -------
+    App
```

Comment:
In `deconfig` terminology, the returned object is an `App`, right? 

I would prefer `App` rather than `new` as the constructor. (This method makes a `new` ...what exactly?)

Reply:
in order to store it internally, we need it to be a function, not a class, so I think we need the lowercase.  My thinking was that users will almost always do `import jdaviz; jdaviz.new()`... so its a new "jdaviz".  But I'm open to `jdaviz.app()` or `jdaviz.new_app()`

Reply:
We're actively emulating the mpl API, where the call would be `plt.figure` which creates a `Figure`, so I'd be in favor of `jdaviz.app`.

Reply:
is there any risk of the user confusing `gca()` and `app()` and calling `app()` to try to get a current instance?